### PR TITLE
Generate an in-transit tree

### PR DESCRIPTION
**Pull request by chee**
all works well. one remaining question:

do we need a **runtime** way of distinguishing one tree from another?

does the `body` node need to have a `transit?: boolean` field? 

i don't think it does... i don't think we need that.

if we generate the `[insert whatever we use to validate on c&m]` types from the `ContentTree.transit` namespace then we're gucci, right?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -266,7 +317,7 @@ doesn't. The text is taken from elsewhere in the article.
 interface ImageSet extends Node {
 	type: "image-set"
 	id: string
-	picture?: ImageSetPicture
+	external picture: ImageSetPicture
```
Comment (chee) [inline README.md:320]: here we mark the picture as `external`

File: tools/maketypes/index.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+const doc = fromMarkdown(Buffer.concat(chunks))
+const code = tangle(doc).replace(
+	/^(interface|type)/gm,
+	term => `export ${term}`
+)
+
+// in the full tree, externals are mandatory
+const full = code.replace(/^(\s+)external (.+)$/gm, "$1$2")
+// in the transit tree, externals must not be present
+const transit = code.replace(/^\s+external (.+:).+$/gm, "")
```
Comment (chee) [inline tools/maketypes/index.js:21]: this is the entire logic for how the two namespaces are generated. remove the `external` modifier for the full tree, remove the entire property for the transit tree

File: README.md [hunk 3]
```diff
@@ -19,7 +19,14 @@ content as an abstract tree.  It implements the **[unist][unist]** spec.
...
 This document defines a format for representing Financial Times article content
-as a tree.  This specification is written in a [typescript][typescript] grammar.
+as a tree. This specification is written in a
+[typescript][https://www.typescriptlang.org/]-like grammar, augmented by the
+addition of the `external` property modifier.
+
+The `external` property modifier indicates that the specified field is absent
+when the `content-tree` is in
+[**transit**](#what-does-it-mean-to-be-in-transit), and required when the
+**content-tree** is at rest.
```
Comment (chee) [inline README.md:29]: this describes the new `external` property modifer. which works like the `readonly` property modifier in typescript, except it only exists in this readme

**Comment by chee** [comment]
message from chee on march 14th 2023: i think we really do need to do the Transit/Full content-tree split. i’m going to do it tomorrow

lol. today is March 15th.

**Review by adgad — APPROVED** [review]
Neat!

I don't think we need any form of runtime validation in the spec IMO. If consumers wish to, they can use something like [`zod`](https://github.com/colinhacks/zod) to validate against the generated types. Customer Products currently does that for the entire C&M response, so it should work for free once they designate `bodyTree` of having type `ContentTree.transit.body`.

Probably want a 👍 from one of the other teams here as well

**Pull request merged**